packages/cli/src/util/get-latest-version/get-latest-worker.js

```diff
@@ -14,36 +14,101 @@ const fetch = require('node-fetch');
 const fs = require('fs-extra');
 const path = require('path');
 const { Agent: HttpsAgent } = require('https');
+const { bold, gray, red } = require('chalk');
+const { format, inspect } = require('util');
 
-process.on('unhandledRejection', err => {
-  if (process.connected) {
-    console.error('Exiting update worker due to error:');
-    console.error(err);
+/**
+ * An simple output helper which accumulates error and debug log messages in
+ * memory for potential persistance to disk while immediately outputting errors
+ * and debug messages, when the `--debug` flag is set, to `stderr`.
+ */
+class WorkerOutput {
+  debugLog = [];
+  logFile = null;
+
+  constructor({ debug = true }) {
+    this.debugOutputEnabled = debug;
+  }
+
+  debug(...args) {
+    this.print('debug', args);
+  }
+
+  error(...args) {
+    this.print('error', args);
+  }
+
+  print(type, args) {
+    const str = format(
+      ...args.map(s => (typeof s === 'string' ? s : inspect(s)))
+    );
+    this.debugLog.push(`[${new Date().toISOString()}] [${type}] ${str}`);
+    if (type === 'debug' && this.debugOutputEnabled) {
+      console.error(
+        `${gray('>')} ${bold('[debug]')} ${gray(
+          `[${new Date().toISOString()}]`
+        )} ${str}`
+      );
+    } else if (type === 'error') {
+      console.error(`${red(`Error:`)} ${str}`);
+    }
   }
+
+  setLogFile(file) {
+    if (this.logFile === null) {
+      process.on('exit', () => {
```

Comment:
**suggestion (non-blocking):** use `process.once` instead

It looks like the check for `this.logFile === null` is used here to ensure that the exit handler is only registered once. It took me a minute to realize that, though. To clarify, could we do one of the following?

- add a comment explaining the intent of the condition
- store the condition value in a variable named with intent, like `isFirstCall`
- remove the condition entirely and use `process.once('exit', ...`

Reply:
`process.once()` won't work here because it's not a matter of the event being fired more than once, but rather we only want to wire up the exit handler once.

No need to define a variable when a comment will work.